src/main/java/com/jmatio/types/MLStructureObjectBase.java

```diff
@@ -42,7 +41,7 @@ protected MLStructureObjectBase(String name, int[] dims, int type, int attribute
 		super(name, dims, type, attributes);
 
 		mlStructArray = new TreeMap<Integer, Map<String, MLArray>>();
-		keys = new LinkedHashSet<String>();
+		keys = new TreeSet<String>();
```

Comment:
In order to preserve key order, it is important for `keys` to remain a `LinkedHashSet<>`.